### PR TITLE
[curl] Refactor curl port

### PR DESCRIPTION
**Pull request by talregev**
Refactor curl port:
- Add test
- Refine dependencies patch
- gssapi feature is not for windows

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by talregev** [comment]
Maybe I need to unify the gssapi and the sspi feature to one feature call: ssapi, and for windows platform should install sspi, and non windows platform it should install gssapi. What do you think?

**Comment by talregev** [comment]
@WangWeiLin-MV Why you change scnlib port on this PR?
May I convert the changes?

**Comment by talregev** [comment]
> @WangWeiLin-MV Why you change scnlib port on this PR? May I revert your changes?

I saw this changes already on the master. I rebase to the master, and your changes was not included. 
I guess it was by mistake.

**Review by MonicaLiu0311 — DISMISSED** [review]

**Comment by MonicaLiu0311** [comment]
Feature `gssapi` is tested successfully in the following triplet:
```
x64-linux
```

File: ports/curl/dependencies.patch [hunk 1]
```diff
@@ -74,16 +74,16 @@ index 049ac34..cc0fe19 100644
    if(LIBSSH2_FOUND)
      list(APPEND CURL_LIBS ${LIBSSH2_LIBRARY})
      list(APPEND CMAKE_REQUIRED_INCLUDES "${LIBSSH2_INCLUDE_DIR}")
-@@ -917,6 +917,14 @@ option(CURL_USE_GSSAPI "Use GSSAPI implementation (right now only Heimdal is sup
+@@ -917,6 +925,14 @@ option(CURL_USE_GSSAPI "Use GSSAPI implementation (right now only Heimdal is sup
  mark_as_advanced(CURL_USE_GSSAPI)
  
  if(CURL_USE_GSSAPI)
 +  find_package(PkgConfig REQUIRED)
```
Comment (dg0yt) [inline ports/curl/dependencies.patch:81]: PKGCONFIG isn't acquired  yet for this feature. But it is necessary.

**Review by dg0yt — CHANGES_REQUESTED** [review]

File: ports/curl/dependencies.patch (hunk 1 above)
Comment (talregev) [reply ports/curl/dependencies.patch:81]: I added to portfile.

**Pull request merged**